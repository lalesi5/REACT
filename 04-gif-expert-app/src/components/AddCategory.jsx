import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({onNewValue}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        //console.log(target.value);
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        
        event.preventDefault();
        if(inputValue.trim().length <= 1) return;

        //setCategories( categories =>  [inputValue, ...categories]);
        onNewValue(inputValue.trim());
        setInputValue('');

    }

    return (

        // <form onSubmit={(event) => onSubmit(event)}>
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                placeholder="Buscar gifts"
                value={inputValue}
                //onChange={(event) => onInputChange(event)}
                onChange={onInputChange}
            />
        </form>

    )
}

AddCategory.propTypes = {
    onNewValue: PropTypes.func.isRequired,
}
