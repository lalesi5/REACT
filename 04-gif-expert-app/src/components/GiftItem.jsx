import React from 'react'

export const GiftItem = ({title,url}) => {
    console.log(title,url);
    return (
        <div className='card'>
            <img src={url} alt={title} />
        </div>
    )
}
