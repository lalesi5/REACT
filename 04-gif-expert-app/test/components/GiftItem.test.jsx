import { render, screen } from "@testing-library/react"
import { GiftItem } from "../../src/components/GiftItem"


describe('Pruebas en <GiftItem/>', () => {

    const title = 'Lales';
    const url = 'https://askldjalkd.com/asd.jpg'

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GiftItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe demostrar la imagen con el URL y ALT indicado', () => {
        render(<GiftItem title={title} url={url} />);
        //screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        //console.log(screen.getByRole('img').src);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('debe de mostrar el titulo en el componente', () => {

        render(<GiftItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();

    })

})