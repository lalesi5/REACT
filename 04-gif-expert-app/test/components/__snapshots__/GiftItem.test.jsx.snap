// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Lales"
      src="https://askldjalkd.com/asd.jpg"
    />
    <p>
      Lales
    </p>
  </div>
</div>
`;
